source-code/sdk-js/src/helpers/ast.test.ts

```diff
@@ -0,0 +1,215 @@
+import { describe, expect, test } from "vitest"
+import { parse } from "acorn"
+import { wrapVariableDeclaration, insertAst } from "./ast.js"
+// @ts-ignore
+import { generate } from "astring"
+import type { ArrowFunctionExpression, Program } from "estree"
+import type { Options } from "acorn"
+
+//@ivanhofer - is ecmaVersion 2020 correct here?
+const acornOptions = {
+	ecmaVersion: 2020,
+	sourceType: "module",
+} as Options
+
+describe("ast - wrapVariableDeclaration", () => {
+	test("Wraps a simple arrow function declaration", async () => {
+		const input = `
+            export const load = () => {
+                return { data: true }
+            }
+        `
+		const expected = `
+            export const load = wrapFn(() => {
+                return { data: true }
+            })
+        `
+		const inputAst = parse(input, acornOptions) as unknown as Program
+		const resultAst = await wrapVariableDeclaration(inputAst, "load", "wrapFn")
+		const expectedAst = parse(expected, acornOptions) as unknown as Program
+		const expectedCode = generate(expectedAst)
+		const resultCode = generate(resultAst)
+		expect(resultCode).toEqual(expectedCode)
+	})
+	test("Return error for nonexistent declarator", async () => {
+		const input = `
+            export const load = () => {
+                return { data: true }
+            }
+        `
+		// @ivanohofer - what do you think of the type casting below?
```

Comment:
`acorn` and `estree` have incompatible types. Can we find a package that provides a `parse` function that is compatible with `estree`?

Reply:
Not sure honestly... Even the babel parser deviates from ESTree: https://babeljs.io/docs/babel-parser#output

Reply:
let'S not waste time on this for now.  Write a TODO comment and use casts for now